c-deps/libroach/encoding_test.cc

```diff
@@ -0,0 +1,69 @@
+// Copyright 2017 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied.  See the License for the specific language governing
+// permissions and limitations under the License.
+
+#include <cinttypes>
+#include <cstdint>
+#include <gtest/gtest.h>
+#include <random>
+#include <vector>
+#include "encoding.h"
+
+TEST(Libroach, Encoding) {
+  // clang-format off
+  std::vector<uint32_t> cases32{
+    0, 1, 2, 3,
+    1 << 16, (1 << 16) - 1, (1 << 16) + 1,
+    1 << 24, (1 << 24) - 1, (1 << 24) + 1,
+    UINT32_MAX, UINT32_MAX - 1,
+  };
+
+  std::vector<uint64_t> cases64{
+    (1ULL << 32) + 1,
+    1ULL << 40, (1ULL << 40) - 1, (1ULL << 40) + 1,
+    1ULL << 48, (1ULL << 48) - 1, (1ULL << 48) + 1,
+    1ULL << 56, (1ULL << 56) - 1, (1ULL << 56) + 1,
+    UINT64_MAX - 1, UINT64_MAX
+  };
+  // clang-format on
```

Comment:
If you're bored, would be nice to let clang format handle these, but it totally mangles `cases64` IMO.